File: src/main/java/no/novari/flyt/archive/gateway/dispatch/web/CreatedLocationPollTimeoutException.java
package no.novari.flyt.archive.gateway.dispatch.web;

import java.net.URI;
import java.time.Duration;

public class CreatedLocationPollTimeoutException extends RuntimeException {

    public CreatedLocationPollTimeoutException() {
        super("Reached max total timeout for polling created location from destination");
    }

    public CreatedLocationPollTimeoutException(URI statusUri, Duration totalTimeout) {
        super("Reached max total timeout for polling created location from destination: statusUri="
              + statusUri + " totalTimeout=" + totalTimeout);
    }

    public CreatedLocationPollTimeoutException(URI statusUri, Duration totalTimeout, String lastStatus) {
        super("Reached max total timeout for polling created location from destination: statusUri="
              + statusUri + " totalTimeout=" + totalTimeout + " lastStatus=" + lastStatus);
    }

}

File: src/main/kotlin/no/novari/flyt/archive/gateway/dispatch/web/CreatedLocationPollTimeoutException.kt
package no.novari.flyt.archive.gateway.dispatch.web

import java.net.URI
import java.time.Duration

class CreatedLocationPollTimeoutException : RuntimeException {
    constructor() : super("Reached max total timeout for polling created location from destination")

    constructor(
        statusUri: URI,
        totalTimeout: Duration,
    ) : super(
        "Reached max total timeout for polling created location from destination: statusUri=$statusUri totalTimeout=$totalTimeout",
    )

    constructor(
        statusUri: URI,
        totalTimeout: Duration,
        lastStatus: String?,
    ) : super(
        "Reached max total timeout for polling created location from destination: " +
            "statusUri=$statusUri totalTimeout=$totalTimeout lastStatus=$lastStatus",
    )
}
